File: clients/CSharp/Model/Player.cs
namespace Aicup2020.Model
{
    public struct Player
    {
        public int Id { get; set; }
        public int Score { get; set; }
        public int Resource { get; set; }
        public Player(int id, int score, int resource)
        {
            this.Id = id;
            this.Score = score;
            this.Resource = resource;
        }
        public static Player ReadFrom(System.IO.BinaryReader reader)
        {
            var result = new Player();
            result.Id = reader.ReadInt32();
            result.Score = reader.ReadInt32();
            result.Resource = reader.ReadInt32();
            return result;
        }
        public void WriteTo(System.IO.BinaryWriter writer)
        {
            writer.Write(Id);
            writer.Write(Score);
            writer.Write(Resource);
        }
    }
}

File: clients/FSharp/Model/Player.fs
#nowarn "0058"
namespace Aicup2020.Model
type Player = {
    Id: int;
    Score: int;
    Resource: int;
    } with
    member this.writeTo(writer: System.IO.BinaryWriter) =
        writer.Write this.Id
        writer.Write this.Score
        writer.Write this.Resource
    static member readFrom(reader: System.IO.BinaryReader) = {
        Id = reader.ReadInt32()
        Score = reader.ReadInt32()
        Resource = reader.ReadInt32()
    }
